src/main/java/io/supertokens/webserver/api/multitenancy/ListTenantsAPI.java

```diff
@@ -63,7 +64,18 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
             boolean shouldProtect = shouldProtectProtectedConfig(req);
             for (TenantConfig tenantConfig : tenantConfigs) {
                 JsonObject tenantConfigJson = tenantConfig.toJson(shouldProtect,
-                        storage, CoreConfig.PROTECTED_CONFIGS);
+                        storage, CoreConfig.PROTECTED_CONFIGS, getVersionFromRequest(req).lesserThan(SemVer.v5_1));
+
+                if (getVersionFromRequest(req).lesserThan(SemVer.v5_1)) {
+                    tenantConfigJson.remove("useFirstFactorsFromStaticConfigIfEmpty");
+                    tenantConfigJson.get("thirdParty").getAsJsonObject().remove("useThirdPartyProvidersFromStaticConfigIfEmpty");
+                }
+
+                if (getVersionFromRequest(req).lesserThan(SemVer.v5_0)) {
+                    tenantConfigJson.remove("firstFactors");
+                    tenantConfigJson.remove("requiredSecondaryFactors");
+                }
```

Comment:
you should make the toJson function only do this logic? Otherwise it is easy to miss doing this

Reply:
Done